src/component/component.ts

```diff
@@ -317,57 +326,57 @@ export class Component<Props extends {} = any, T extends Env = Env> {
    * Note that a component can be mounted an unmounted several times
    */
   async mount(target: HTMLElement | DocumentFragment, options: MountOptions = {}): Promise<void> {
+    if (!(target instanceof HTMLElement || target instanceof DocumentFragment)) {
+      let message = `Component '${this.constructor.name}' cannot be mounted: the target is not a valid DOM node.`;
+      message += `\nMaybe the DOM is not ready yet? (in that case, you can use owl.utils.whenReady)`;
+      throw new Error(message);
+    }
     const position = options.position || "last-child";
     const __owl__ = this.__owl__;
-    if (__owl__.isMounted) {
-      if (position !== "self" && this.el!.parentNode !== target) {
-        // in this situation, we are trying to mount a component on a different
-        // target. In this case, we need to unmount first, otherwise it will
-        // not work.
-        this.unmount();
-      } else {
-        return Promise.resolve();
+    const currentFiber = __owl__.currentFiber;
+
+    switch (__owl__.status) {
+      case STATUS.CREATED: {
+        const fiber = new Fiber(null, this, true, target, position);
+        fiber.shouldPatch = false;
+        this.__prepareAndRender(fiber, () => {});
+        return scheduler.addFiber(fiber);
       }
-    }
-    if (__owl__.isDestroyed) {
-      throw new Error("Cannot mount a destroyed component");
-    }
-    if (__owl__.currentFiber) {
-      const currentFiber = __owl__.currentFiber;
-      if (!currentFiber.target && !currentFiber.position) {
-        // this means we have a pending rendering, but it was a render operation,
-        // not a mount operation. We can simply update the fiber with the target
-        // and the position
+      case STATUS.WILLSTARTED:
+      case STATUS.STARTED:
         currentFiber.target = target;
         currentFiber.position = position;
         return scheduler.addFiber(currentFiber);
```

Comment:
not sure about this: are we sure there is a currentFiber? so `started` means there is a pending rendering?

Reply:
So `ready` means rendering is done, but not in the DOM?